Training/Training/TestReversal.c

```diff
@@ -0,0 +1,131 @@
+// TestReversal.c
+// C program to check test cases
+// ------------------------------------------------------------------------------------------------
+
+#include "Reversal.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/// <summary>Function to get string input from user</summary>
+void GetStringInput ();
+/// <summary>Function to get integer input from user</summary>
+void GetIntegerInput ();
+/// <summary>Function to print result</summary>
+void PrintResult (int value);
+/// <summary>Function to print result along with reversed number</summary>
+void PrintResultInt (int value, long long int num);
+/// <summary>Function to check test cases</summary>
+void OutputCheck ();
+/// <summary>Clears the content on screen</summary>
+void ClrScreen ();
+
+int main () {
+   while (1) {
+      char ch = '\n';
+      char option[3];
+      printf ("1-PALINDROME CHECKER\n2-REVERSE A NUMBER\n3-TEST CASES\n4-CLEAR SCREEN\n5-EXIT\nEnter an option:");
+      fgets (option, sizeof (option), stdin);
+      char* Result = strchr (option, ch);
+      if (Result == NULL) {
+         while (getchar () != '\n');
+         printf (ANSI_COLOR_RED"Please enter number between 1-5\n\n"ANSI_RESET_ALL);
+      } else {
+         int num = atoi (option);
+         switch (num) {
+            case 1:
+               GetStringInput ();
+               break;
+            case 2:
+               GetIntegerInput ();
+               break;
+            case 3:
+               OutputCheck ();
+               break;
+            case 4:
+               ClrScreen ();
+               break;
+            case 5:
+               return 0;
+               break;
+         }
+      }
+   }
+   return 0;
+}
+
+void GetStringInput () {
+   char ch = '\n';
+   char phrase[MAX_CHAR];
+   printf ("\nEnter a Phrase: ");
+   char* j = fgets (phrase, sizeof (phrase), stdin);
+   char* strResult = strchr (phrase, ch); //To check whether input exceeds limit
+   if (strResult == NULL) while (getchar () != '\n');
+   (j != NULL && phrase[0] != '\n' && strResult != NULL) ? PrintResult (StrIsPalindrome (phrase, 1)) : printf (ANSI_COLOR_RED"INVALID!!!\n\n"ANSI_RESET_ALL);
+}
+
+void GetIntegerInput () {
+   errno = 0;
+   char ch = '\n';
+   char input[MAX_DIGIT];
+   int number = 0;
+   char* endptr;
+   printf ("\nEnter an integer: ");
+   char* i = fgets (input, sizeof (input), stdin);
+   char* intResult = strchr (input, ch); //To check whether input exceeds limit
+   number = strtol (input, &endptr, 10);
+   long long int revNum = 0;
+   long long int* ptr = &revNum;
+   if (intResult == NULL) while (getchar () != '\n');
+   (i == NULL || endptr == input || *endptr != '\n' || input[0] == ' ' || input[0] == '\t') ? printf (ANSI_COLOR_RED"INVALID!!!\n\n"ANSI_RESET_ALL) :
+      (errno == ERANGE) ? printf (ANSI_COLOR_RED"INPUT LIMIT EXCEEDED!!!\n\n"ANSI_RESET_ALL) : PrintResultInt (NumIsPalindrome (number, ptr), revNum);
+}
+
+void PrintResult (int value) {

```

Comment:
This function is not needed since it has only one line. Apply to the other print function also.